plugins/languages/treesitter/treesitter.nix

```
@@ -6,210 +6,317 @@
-        type = with types; attrsOf (either str (listOf str));
-        description = ''
-          This is a wrapping of the `vim.treesitter.language.register` function.
-          Register specific parsers to one or several filetypes.
-          The keys are the parser names and the values are either one or several filetypes.
-        '';
-        default = { };
-        example = {
-          cpp = "onelab";
-          python = [
-            "myFiletype"
-            "anotherFiletype"
-          ];
-        };
-      };
+    ignore_install = helpers.defaultNullOpts.mkListOf types.str [ ] ''
+      List of parsers to ignore installing (for "all")
+    '';
+
+    parser_install_dir = helpers.defaultNullOpts.mkStr "$XDG_DATA_HOME/nvim/treesitter" ''
+      Location of the parsers to be installed by the plugin (only needed when `nixGrammars` is disabled).
+
+      This default might not work on your own install, please make sure that `$XDG_DATA_HOME` is set if you want to use the default. Otherwise, change it to something that will work for you!
+    '';
+
+    sync_install = helpers.defaultNullOpts.mkBool false ''
+      Install parsers synchronously (only applied to `ensure_installed`).
+    '';
+  };
+
+  settingsExample = {
+    additional_vim_regex_highlighting = true;
+    ensure_installed = true;
+    ignore_install = [ "rust" ];
+    indent = true;
+    parser_install_dir = "$XDG_DATA_HOME/nvim/treesitter";
+
+    highlight = {
+      enable = true;
+      disable = [ "rust" ];
+      custom_captures = { };
+    };
 
-      grammarPackages = mkOption {
-        type = with types; listOf package;
-        default = cfg.package.passthru.allGrammars;
-        description = "Grammar packages to install";
+    incremental_selection = {
+      enable = true;
+
+      keymaps = {
+        init_selection = "gnn";
+        node_incremental = "grn";
+        scope_incremental = "grc";
+        node_decremental = "grm";
       };
+    };
+  };
+
+  extraOptions = {
+    folding = mkOption {
+      type = types.bool;
+      default = true;
+      example = false;
+      description = "Whether to enable tree-sitter based folding.";
+    };
+
+    gccPackage = helpers.mkPackageOption {
+      name = "gcc";
+      default = pkgs.gcc;
+      defaultText = literalExpression "pkgs.gcc";
+      example = literalExpression "pkgs.gcc14";
+      description = ''
+        Which package (if any) to be added as the GCC compiler.
+        This is required to build grammars if you are not using `nixGrammars`.
+        To disable the installation of GCC, set this option to `null`.
+      '';
+    };
 
-      moduleConfig = mkOption {
-        type = types.attrsOf types.anything;
-        default = { };
-        description = "This is the configuration for extra modules. It should not be used directly";
+    grammarPackages = mkOption {
+      type = with types; listOf package;
+      default = config.plugins.treesitter.package.passthru.allGrammars or [ ];
```

```suggestion
      default = config.plugins.treesitter.package.passthru.allGrammars;
```
Turns out this wasn't needed